Made Risk Free Rate no longer hardcoded

northern-64bit opened the pull request:
Replaced hardcoded rfr with a function, which returns historical rfr from the API endpoint of St. Louis Fed Data Source FRED. Fixing: Issue #46

northern-64bit committed:
Update init.edn

northern-64bit committed:
Update init.edn

northern-64bit committed:
Update communication.clj

northern-64bit committed:
Fixed error

northern-64bit committed:
Create External APIs.md

A description of the TD Ameritrade API

northern-64bit committed:
Made Risk Free Rate no longer hardcoded

Replaced hardcoded rfr with a function, which returns historical rfr from the API endpoint of St. Louis Fed Data Source FRED.

Tyruiop commented:
This is a good start, however it is not enough. 

The function you created allows to get the RFR (great), however we want to automatically get it every morning. Esssentially, you want to write a full new crawler just for this value :).

Tyruiop commented:
To simplify your task, I merged https://github.com/Tyruiop/syncretism/pull/48, you should rebase on top of `main` (if you are not sure how to do it, I can walk you through it), and then you can follow the same structure as `fundamentals.clj`, except that you want it to crawl once a day :).

northern-64bit committed:
Merge branch 'Tyruiop:main' into patch-1

northern-64bit committed:
Added RFR Cralwer

- Added RFR Crawler in crawler.shared.clj
- Updated README.md from mentioning  dataops-crawler to syncretism-crawler

Tyruiop commented on syncretism-crawler/src/syncretism/crawler/shared.clj:
```diff
@@ -5,20 +5,55 @@
     :fundamentals-status :paused
-    :options-status :running}))
+    :options-status :running
+    :rfr-status :paused}))
```
Should start `:running`, we want it constantly running

Tyruiop commented on syncretism-crawler/src/syncretism/crawler/shared.clj:
```diff
@@ -5,20 +5,55 @@
+           (println (format "Gathering the Risk Free Rate"))
+           (try
+             (let [data (rfr)]
+                  (json/write-str data))
```
you are reading and writing the same json here, no need. You want to extract the correct value, and replace the current - in memory - value. You probably want to use an `atom` for that :)

Also, you want this crawler to start before the options crawler, because we need the RFR to be saved while options are crawled.

northern-64bit replied:
I think that it's fixed now.

Tyruiop reviewed (changes requested):
Good progress!

northern-64bit committed:
Fixed Requested Changes

Tyruiop commented:
Last thing to do!

https://github.com/Tyruiop/syncretism/blob/main/syncretism/src/syncretism/greeks.clj#L85

This function should take `:rfr` as parameter in the input map when it's called (check where in `options.clj`), and propagate this input to the needed functions (you'll see the `(def rfr ...)` on top, just find where it is.

Then to test you can just do `lein install` in the `syncretism/` folder and `lein run` in `syncretism-crawler`

northern-64bit committed:
Update greeks.clj

I think that I fixed it.

northern-64bit commented:
> Last thing to do!
> 
> https://github.com/Tyruiop/syncretism/blob/main/syncretism/src/syncretism/greeks.clj#L85
> 
> This function should take `:rfr` as parameter in the input map when it's called (check where in `options.clj`), and propagate this input to the needed functions (you'll see the `(def rfr ...)` on top, just find where it is.
> 
> Then to test you can just do `lein install` in the `syncretism/` folder and `lein run` in `syncretism-crawler`

I tried to do what you described in my´most recent commit and replaced rfr with the new crawler variable risk-free-rate, but it cannot find it. Should do that risk-free-rate is a global variable or what am I missing?
